### ISSUE TITLE
Uploader crash

### ISSUE THREAD

**Issue by q1p** [cloudinary/cloudinary_java#34]
This crash is caused by specific file name: "Universal Image Loader @#&=+-_.,!()~'%20.png"

```
 java.lang.StringIndexOutOfBoundsException: length=44; regionStart=41; regionLength=-17
     at java.lang.String.startEndAndLength(String.java:504)
     at java.lang.String.substring(String.java:1333)
     at java.net.DefaultFileNameMap.getContentTypeFor(DefaultFileNameMap.java:38)
     at java.net.URLConnection.guessContentTypeFromName(URLConnection.java:680)
     at com.cloudinary.android.MultipartUtility.addFilePart(MultipartUtility.java:91)
     at com.cloudinary.android.MultipartUtility.addFilePart(MultipartUtility.java:85)
     at com.cloudinary.android.UploaderStrategy.callApi(UploaderStrategy.java:67)
     at com.cloudinary.Uploader.callApi(Uploader.java:22)
     at com.cloudinary.Uploader.upload(Uploader.java:55)
```

**Comment by wagaun** [comment]
Hi, I'm trying to fix this issue. First of all I tried to write a test case in UploaderTest to replicate it. But I'm trying to run it in Android Studio and I'm failing.

Here what I did so far:
-Created a method to replicate the bug behaviour in the UploaderTest class
-Changed the AndroidManifest.xml.sample file with my cloudinary credentials and created an AndroidManifest.xml with the same content in the same path
-Tried to run the Test

When I'm going to run the test case a "Edit configuration" dialog appears with the following message at the bottom: "Error: AndroidManifest.xml doesn't exist or has incorrect root tag". If I insist and click to run an emulator is opened and I don't know what to do next.

Sorry about this huge description. I think I missed something here.

**Comment by wagaun** [comment]
Could not replicate the bug.

Added the following method in AbstractUploaderTest class and also a PNG file with the name as described, and it seems to work fine for me.
```
 @Test
 public void testUploadSpecificFilename() throws IOException {
        JSONObject result = new JSONObject(cloudinary.uploader().upload("Universal Image Loader @#&=+-_.,!()~_%20.png",
                ObjectUtils.emptyMap()));
}
```

**Comment by TalLevAmi** [comment]
The issue was fixed in version 1.2.2. We missed the reference to this issue.
